openmdao/drivers/doe_driver.py

```diff
@@ -116,23 +117,19 @@ def _setup_comm(self, comm):
         MPI.Comm or <FakeComm> or None
             The communicator for the Problem model.
         """
-        parallel = self.options['parallel']
-        if MPI and parallel:
+        if MPI and self.options['run_parallel']:
             self._comm = comm
-
-            if parallel == 1:  # True == 1
-                size = 1
-                color = self._color = comm.rank
-            else:
-                comm_size = comm.size
-                size = comm_size // parallel
-                if comm_size != size * parallel:
-                    raise RuntimeError("The number of processors is not evenly divisable by "
-                                       "the specified number of parallel cases.\n Provide a "
-                                       "number of processors that is a multiple of %d, or "
-                                       "specify a number of parallel cases that divides "
-                                       "into %d." % (parallel, comm_size))
-                color = self._color = comm.rank % size
+            procs_per_model = self.options['procs_per_model']
+
+            full_size = comm.size
+            size = full_size // procs_per_model
+            if full_size != size * procs_per_model:
+                raise RuntimeError("The total number of processors is not evenly divisable by the "
```

Comment:
divisable --> divisible

Reply:
global search and replace of all instances